grades/api.py

```diff
@@ -230,3 +232,31 @@ def generate_program_certificate(user, program):
         user.username,
         program.title
     )
+
+
+def update_or_create_combined_final_grade(user, course):
+    """
+    Update or create CombinedFinalGrade
+
+    Args:
+        user (User): a django User
+        course (Course): a course model object
+    """
+    mmtrack = get_mmtrack(user, course.program)
+    final_grade = mmtrack.get_best_final_grade_for_course(course)
+    if final_grade is None:
+        log.error('User [%s] does not have a final for course [%s]', user, course)
+        return
+
+    best_exam = mmtrack.get_best_proctored_exam_grade(course)
+    if best_exam is None:
+        log.error('User [%s] does not have a passing exam grade for course [%s]', user, course)
```

Comment:
Sorry just noticed this, should these be `warn` instead? I don't think we want to get sentry errors every time one of the signals trigger but we decide not to create a `CombinedFinalGrade`